addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -2893,6 +2894,186 @@ QUnit.test('post a simple message', async function (assert) {
     );
 });
 
+QUnit.test('post message on non-mailing channel with "Enter" keyboard shortcut', async function (assert) {
+    assert.expect(2);
+
+    // channel expected to be found in the sidebar
+    // with a random unique id that will be referenced in the test
+    this.data['mail.channel'].records.push({ id: 20, mass_mailing: false });
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.channel_20',
+            },
+        },
+        async mockRPC(route, args) {
+            const res = await this._super(...arguments);
+            if (args.method === 'message_post') {
+                assert.step('message_post');
+            }
+            return res;
+        },
+    });
+
+    // insert some HTML in editable
+    await afterNextRender(() => {
+        document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+        document.execCommand('insertText', false, "Test");
+    });
+    await afterNextRender(() => {
+        const kevt = new window.KeyboardEvent('keydown', { key: "Enter" });
+        document.querySelector('.o_ComposerTextInput_textarea').dispatchEvent(kevt);
+    });
+    assert.verifySteps(
+        ['message_post'],
+        "should have posted message on pressing 'Enter' in text input of composer"
+    );
+});
+
+QUnit.test('do not post message on non-mailing channel with "SHIFT-Enter" keyboard shortcut', async function (assert) {
+    assert.expect(1);
+
+    // channel expected to be found in the sidebar
+    // with a random unique id that will be referenced in the test
+    this.data['mail.channel'].records.push({ id: 20, mass_mailing: true });
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.channel_20',
+            },
+        },
+        async mockRPC(route, args) {
+            const res = await this._super(...arguments);
+            if (args.method === 'message_post') {
+                assert.step('message_post');
+            }
+            return res;
+        },
+    });
+
+    // insert some HTML in editable
+    await afterNextRender(() => {
+        document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+        document.execCommand('insertText', false, "Test");
+    });
+    await afterNextRender(() => {
+        const kevt = new window.KeyboardEvent('keydown', { key: "Enter", shiftKey: true });
+        document.querySelector('.o_ComposerTextInput_textarea').dispatchEvent(kevt);
+    });
+    assert.verifySteps(
+        [],
+        "should not have posted message on pressing 'SHIFT-Enter' in text input of composer"
```

Comment:
Shouldn't test be about shift-enter inserting new line instead?

Reply:
For some reasons, simulated shift-enter keydown doesn't newline. I have no idea why...

Reply:
I'm unable to simulate newline on textarea from ENTER key press.

Some default browser behaviour seems to add line break on value of text area (visible in `_onInputTextarea`), but I can't trigger this browser default code with simulated keyboard events (even tried with reusing genuine keydown event for simulated one, in vain...).

The closest way to simulate is to do something like:
```js
// - textarea.value += '\r\n'; (from default browser behaviour)
// - trigger keydown enter
// - trigger input (trigger re-render/store sync/auto-height compute of textarea)
```
But I'm not sure if that's acceptable to have this in test. What do you think?